sdk/core/azure-core/azure/core/polling/async_base_polling.py

```diff
@@ -43,10 +43,17 @@ class AsyncLROBasePolling(LROBasePolling):
     async def run(self):  # pylint:disable=invalid-overridden-method
         try:
             await self._poll()
+        except AzureError as err:
+            if not err.continuation_token:
+                err.continuation_token = self.get_continuation_token()
+            raise
+
         except BadStatus as err:
             self._status = "Failed"
             raise HttpResponseError(
-                response=self._pipeline_response.http_response, error=err
+                response=self._pipeline_response.http_response,
+                error=err,
+                continuation_token=self.get_continuation_token()
```

Comment:
If get_continuation_token raise an exception somehow, I feel we should stick to HttpResponseError. IOW, if the current is unstable enough (after all we know an exception is being raised), we should be sure we raise the initial exception, and not a "Not able to build continuation token" exception